### PR TITLE
Ke/add loading adhoc container

### PR DESCRIPTION
**Pull request by Ke-Jin-Laserfiche**

### CODE REVIEW

File: projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.css [hunk 1]
```diff
@@ -51,27 +51,27 @@ i.loader {
...
 
 .field-picker-container.has-changes {
   bottom: 52px;
 }
 
+.lf-checkbox {
+  height: 25px;
+}
+
 .button-container {
```
Comment (alexgomezlf) [inline projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.css]: .button-container should be more specific

File: projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.css [hunk 2]
```diff
@@ -8,10 +9,9 @@
 }
 
 .pick-field {
```
Comment (alexgomezlf) [inline projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.css]: make .pick-field more specific

File: projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html [hunk 3]
```diff
@@ -8,33 +8,39 @@
...
+        <div class="search">
+            <input [value]="fieldFilterText" [formControl]="filterFieldsControl" placeholder="{{SEARCH_FIELDS | async}}"
+                class="searchBox">
+            <div class="icon icon-search"></div>
+            <div *ngIf="fieldFilterText && fieldFilterText.length > 0" (click)="onClearFields()"
+                class="icon icon-clear">
+            </div>
+        </div>
+        <cdk-virtual-scroll-viewport itemSize="25" class="pick-field">
+          <div *cdkVirtualFor="let field of displayFieldInfos, let i = index">
```
Comment (alexgomezlf) [inline projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html]: did you test the case where there's no fields with the filter?
The cdk-virtual-scroll-viewport will exist, but won't mess with the message?

File: projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html [hunk 4]
```diff
@@ -8,33 +8,39 @@
...
-          <mat-checkbox color="primary" class="lf-checkbox" (change)="onUpdateCheckbox($event, field)" id=`add-remove-checkbox-{{i}}`
-              [checked]="isFieldSelected(field)" title="{{field.name}}" [disabled]="field.inTemplateSelected">
-              {{field.name}} ({{field | getFieldType: LOCALIZED_FIELD_TYPES | async}})
-          </mat-checkbox>
-      </div>
-      <div class="fields-empty" *ngIf="!displayFieldInfos || displayFieldInfos.length===0">
-          {{NO_MATCHING_FIELDS_FOUND | async}}
-      </div>
+      </cdk-virtual-scroll-viewport>
+        <div class="fields-empty" *ngIf="!displayFieldInfos || displayFieldInfos.length===0">
```
Comment (alexgomezlf) [inline projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html]: there is a bug where if the repo has no fields this should say no available fields instead.. Is there a way to tell that? allFieldInfos vs displayed?
We have a separate bug for that, just something to keep in mind

File: projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html (hunk 3 above)
Comment (Ke-Jin-Laserfiche) [reply projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html]: yeah it was messing with the message but I changed it. good catch

Comment (alexgomezlf) [reply projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html]: the solution was to add a class to the no fields message, not hide the overlay?

Comment (Ke-Jin-Laserfiche) [reply projects/ui-components/lf-metadata/lf-field-adhoc-container/lf-field-add-remove/lf-field-add-remove.component.html]: This class makes the no field message takes the full space of the parent container. I will change the name of the class and also hide the overlay

**Review by alexgomezlf — DISMISSED** [review]

**Review by alexgomezlf — APPROVED** [review]

**Pull request merged**